src/main/java/elements/MovableElement.java

```diff
@@ -50,7 +67,7 @@ public boolean processTick() {
     @Override
     public boolean move() {
         Point2D currPos = super.getPos();
-        Point2D newPos = Point2D.add(currPos, this.velocity);
+        Point2D newPos = Point2D.add(currPos, this.getVelocity());
```

Comment:
just use this.velocity instead of the getter

Reply:
For the ChasingElement to follow the Player, its velocity needs to change and we need to get every update instantly. This.velocity does not reflect the change....maybe because it was memorized at instantiation? Not sure, but it did not work.

Reply:
I’ll take a look on my end then.